### PR TITLE
Fix jumpy scrollbars

### PR DESCRIPTION
**Pull request by martenbjork**
## Problem statement

- When a page in the admin loads, we show skeleton content.
- When the actual content come in, it often overflows the window height, causing scrollbars to appear.
- This means that the page suddenly jumps 16px, which feels janky.

## Fix

- Introduce subtle, overlay scrollbars that live on top of the content.
- The scrollbars do not push to the left when the scrollbars appear. 

## Methodology

- On the body, add the legacy `overflow: overlay` property
- On the nav, use the `scrollbar-gutters` property
- Style all scrollbars to be very subtle

Using this property would allow us to reduce jumpiness for Chromium users with minimal effort:

https://dev.to/jonosellier/easy-overlay-scrollbars-variable-width-1mbh

We could then wait for w3 to finally ship support for overlay scrollbars:

https://www.w3.org/TR/css-overflow-3/#overlay-scrollbars

### CODE REVIEW

File: polaris-react/src/styles/shared/_layout.scss [hunk 1]
```diff
@@ -15,31 +15,36 @@
   #{$property}: calc(#{$spacing} + env(safe-area-inset-#{$area}));
 }
 
-@mixin custom-scrollbars() {
-  scrollbar-gutter: stable;
+@mixin custom-scrollbars($onRoot: false) {
+  scrollbar-width: thin;
+
+  @if $onRoot {
```
Comment (mattkubej) [inline polaris-react/src/styles/shared/_layout.scss:22]: I'm wondering if we should consider renaming `onRoot` to `overlay` or `overflowY` to make it more clear what occurs by toggling this value. What are your thoughts?

Comment (martenbjork) [reply polaris-react/src/styles/shared/_layout.scss:22]: I wanted to communicate that this should only be used on the body and nowhere else. However, `onRoot` made more sense when it was used on the `html` element. Now its used on `body`. What do you think about simply calling it `onBody`? That should detract folks from using it anywhere else.

Comment (mattkubej) [reply polaris-react/src/styles/shared/_layout.scss:22]: If that's the intention, then that sounds reasonable to me. I think another approach, where we could avoid the parameter is making explicit mixins to make this clear as well. Maybe something like this?

```scss
@mixin _scrollbar-common {
  scrollbar-width: thin;

  &::-webkit-scrollbar {
    width: 7px;
  }

  &::-webkit-scrollbar-track {
    background: transparent;
  }

  &::-webkit-scrollbar-thumb {
    background: rgba(0, 0, 0, 0);
    border-radius: 100px;
    transition: all 0.2s linear;
  }

  &:hover {
    &::-webkit-scrollbar-thumb {
      background: rgba(0, 0, 0, 0.25);

      &:hover {
        background: rgba(0, 0, 0, 0.5);
      }
    }
  }
}

@mixin custom-scrollbars {
  @include _scrollbar-common;
}

@mixin custom-body-scrollbars {
  @include _scrollbar-common;
  
  @supports #{'\selector(*::-webkit-scrollbar-thumb)'} {
    overflow-y: overlay;
  }
}
```

**Pull request closed without merging**